### PR TITLE
[chore][receiver/mysql]: extract handling global stats to another function

### PR DESCRIPTION
**Pull request by aboguszewski-sumo**
**Description:** 
A follow-up to this discussion: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/14328#discussion_r976521456
This pull request extracts logic behind fetching and handling global stats so that the errors can be handled properly.

### CODE REVIEW

**Comment by djaglowski** [comment]
Please rebase

File: receiver/mysqlreceiver/scraper.go [hunk 1]
```diff
@@ -99,10 +99,83 @@ func (m *mySQLScraper) scrape(context.Context) (pmetric.Metrics, error) {
...
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeInsert/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsInsert, s.name, s.schema, s.index,
+		)
+		m.mb.RecordMysqlIndexIoWaitTimeDataPoint(
+			now, s.timeUpdate/picosecondsInNanoseconds, metadata.AttributeIoWaitsOperationsUpdate, s.name, s.schema, s.index,
+		)
+	}
+}
+
+func (m *mySQLScraper) scrapeGlobalStats(now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
```
Comment (djaglowski) [inline receiver/mysqlreceiver/scraper.go:173]: For the sake of a cleaner diff, will you please move this function definition above the table and index functions?

Comment (aboguszewski-sumo) [reply receiver/mysqlreceiver/scraper.go:109]: I did this. I should have noticed it before, sorry.

File: receiver/mysqlreceiver/scraper.go [hunk 2]
```diff
@@ -99,10 +99,19 @@ func (m *mySQLScraper) scrape(context.Context) (pmetric.Metrics, error) {
...
+
+	return m.mb.Emit(), errs.Combine()
+}
+
+func (m *mySQLScraper) scrapeGlobalStats(now pcommon.Timestamp, errs *scrapererror.ScrapeErrors) {
 	globalStats, err := m.sqlclient.getGlobalStats()
 	if err != nil {
 		m.logger.Error("Failed to fetch global stats", zap.Error(err))
-		return pmetric.Metrics{}, err
+		errs.AddPartial(1, err)
```
Comment (djaglowski) [inline receiver/mysqlreceiver/scraper.go:113]: One more nit - can you update this number to match the number of data points we'll miss if the call fails? (62 by my count, but perhaps you can double check me. I counted 4 from `recordDataPages|Usage`)

Comment (aboguszewski-sumo) [reply receiver/mysqlreceiver/scraper.go:113]: I'm pushing the version with 66 and I'll correct it if you don't agree.

**Review by djaglowski — APPROVED** [review]

**Review by codeboten — APPROVED** [review]

**Pull request merged**